### ISSUE TITLE
remove SPEC in openjdk_tests

### ISSUE THREAD

**Issue by llxia** [AdoptOpenJDK/openjdk-tests#2363]
With TKG auto-detecting SPEC, I think we can remove all SPEC in openjdk_tests. For example, https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/openjdk_tests#L5

This may affect special builds like Valhalla. 

FYI @renfeiw

**Comment by jwnmoon** [comment]
Hello, is this issue still open?

**Comment by smlambert** [comment]
Yes, this issue is still open and valid.  I believe to resolve it, is to remove all references to SPEC found in the https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/openjdk_tests file, and then a reviewer will have to test that it does not break the Valhalla builds.

Shall I assign you @jwnmoon ?

**Comment by jwnmoon** [comment]
Yes, I'll work on it!

**Comment by jeffalder** [comment]
This is the same file I'm editing on #2753; I'm not sure if I should just do this change as well?

**Comment by llxia** [comment]
This one is a little complicated as we also need to update other files (i.e., [JenkinsfileBase](https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase)) to use PLATFORM instead of SPEC. Maybe a separate PR be is better. It will be easier for testing. Also, to avoid merge conflict, I would suggest waiting until #2753 is merged.

**Comment by jeffalder** [comment]
@llxia I was browsing through this. I could replace all of the references to `SPEC.startsWith` to `PLATFORM.contains`, and that should cover the removal of `SPEC`. However, I find that approach somewhat fragile, as it requires that `PLATFORM` always match a certain pattern. That may or may not surprise people adding new platform support. 

I'd like to fix this by replacing `SPEC` with `BASE_OS`; `BASE_OS` has one of the values `('zos','linux','osx','windows','aix','solaris')`. Then all the references to `SPEC.startsWith` become `BASE_OS == `. As a step-wise increment goes, I like this approach better than `PLATFORM.contains`.

Two things - 
1. Is this approach feasible? Do you agree with this approach?
2. Is there a way to test my changes to validate that I didn't break anything? There are some things about variable passing between `load JenkinsfileBase` that I don't quite understand.

**Comment by llxia** [comment]
Thanks @jeffalder for working on this.  Talked with @renfeiw , since there are other dependencies on `PLATFORM_MAP`, we need to approach this progressively. Sorry about the confusion.

For now, we can only remove SPEC references [JenkinsfileBase](https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase). We do not need to update openjdk_tests or any of the build.xml.
For example: 
- https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase#L53

change `env.SPEC.startsWith('zos')` to `PLATFORM.contains('zos')`

- `env.SPEC = "${SPEC}"` ([link](https://github.com/adoptium/aqa-tests/blob/master/buildenv/jenkins/JenkinsfileBase#L42)) is the only exception that cannot be removed.

Note: `PLATFORM` is a special parameter and it has to follow the naming convention.

**Comment by jeffalder** [comment]
Hey, I let this go on too long. I'm not going to complete it, and I'm yielding my hold on this. My apologies!
